frontend/src/api/services/auth.ts

```diff
@@ -0,0 +1,25 @@
+import { errorHandler } from "api/error-handler";
+import { AxiosInstance, AxiosError } from "axios";
+import { UserLogin, User } from "../models/user";
+
+export class AuthService {
+  constructor(private httpClient: AxiosInstance) {
+    this.httpClient = httpClient;
+  }
+
+  public async authenticate(user: UserLogin): Promise<User | undefined> {
+    try {
+      const response = await this.httpClient.post<User>(`/users/sign_in`, {
+        user,
+      });
+
+      if (response && response.data) {
```

Comment:
if this condition returns false what happens?

Reply:
Fixed on error-handler refactor